### PR TITLE
fix(runtime): significant refactor

### PR DESCRIPTION
**Pull request by pmuir**
* no hacks!!!
* global support

### CODE REVIEW

File: src/app/app.module.ts [hunk 1]
```diff
@@ -1,6 +1,4 @@
-import { AnalyticService } from './shared/analytics.service';
-import { ConfigStore } from './base/config.store';
-import { fabric8UIConfigProvider } from './shared/config/fabric8-ui-config.service';
+import { NamespaceScope } from 'fabric8-runtime-console/src/app/kubernetes/service/namespace.scope';
```
Comment (joshuawilson) [inline src/app/app.module.ts]: please move this down alphabetically to the section that is local imports

File: src/app/create/environments/environments-routing.module.ts [hunk 2]
```diff
@@ -1,3 +1,4 @@
+import { OAuthConfigStoreGuard } from './../../shared/runtime-console/oauth-config-store-guard.service';
```
Comment (joshuawilson) [inline src/app/create/environments/environments-routing.module.ts]: please move this down alphabetically to the section that is local imports

File: src/app/create/environments/environments.component.ts [hunk 3]
```diff
@@ -1,26 +1,58 @@
-import { BuildConfigService } from 'fabric8-runtime-console/src/app/kubernetes/service/buildconfig.service';
-import { SwitchableNamespaceScope } from './switchable-namepsace.scope';
-import { Space, SpaceService } from 'ngx-fabric8-wit';
-import { Observable } from 'rxjs';
 import { Component, OnInit } from '@angular/core';
 import { Router } from '@angular/router';
```
Comment (joshuawilson) [inline src/app/create/environments/environments.component.ts]: import not used

File: src/app/create/environments/environments.component.ts [hunk 4]
```diff
@@ -1,26 +1,58 @@
...
+
+import { Observable, ConnectableObservable } from 'rxjs';
```
Comment (joshuawilson) [inline src/app/create/environments/environments.component.ts]: ConnectableObservable import not used

File: src/app/create/environments/environments.component.ts [hunk 5]
```diff
@@ -1,26 +1,58 @@
...
+
 import { AuthenticationService, UserService } from 'ngx-login-client';
```
Comment (joshuawilson) [inline src/app/create/environments/environments.component.ts]: import not used

File: src/app/create/environments/environments.component.ts [hunk 6]
```diff
@@ -1,26 +1,58 @@
...
+import { Space, SpaceService, Spaces, Contexts } from 'ngx-fabric8-wit';
```
Comment (joshuawilson) [inline src/app/create/environments/environments.component.ts]: SpaceService, Spaces import not used

File: src/app/create/environments/environments.component.ts [hunk 7]
```diff
@@ -1,26 +1,58 @@
...
+
+import {
+  ToolbarConfig,
+  FilterConfig,
+  FilterQuery,
+  FilterEvent,
+  Filter,
+  SortEvent,
+  SortField,
+  TreeListItemComponent
```
Comment (joshuawilson) [inline src/app/create/environments/environments.component.ts]: import not used

File: src/app/create/environments/environments.component.ts [hunk 8]
```diff
@@ -1,26 +1,58 @@
...
+} from 'ngx-widgets';
+
+import {
+  TreeComponent,
```
Comment (joshuawilson) [inline src/app/create/environments/environments.component.ts]: import not used

File: src/app/create/environments/environments.component.ts [hunk 9]
```diff
@@ -1,26 +1,58 @@
...
+  TreeNode,
+  TREE_ACTIONS,
+  KEYS,
```
Comment (joshuawilson) [inline src/app/create/environments/environments.component.ts]: import not used

File: src/app/create/environments/environments.module.ts [hunk 10]
```diff
@@ -11,54 +11,39 @@ import {
...
 
+import { TreeModule } from 'angular2-tree-component';
+
+import {
+  KubernetesRestangularModule,
+  NamespaceScope,
+  DeploymentModule,
+  DevNamespaceScope
+} from 'fabric8-runtime-console';
+import { RestangularModule } from 'ng2-restangular';
```
Comment (joshuawilson) [inline src/app/create/environments/environments.module.ts]: import not used

File: src/app/create/environments/environments.module.ts [hunk 11]
```diff
@@ -11,54 +11,39 @@ import {
   TooltipConfig
 } from 'ng2-bootstrap';
 
+import { TreeModule } from 'angular2-tree-component';
+
+import {
+  KubernetesRestangularModule,
+  NamespaceScope,
+  DeploymentModule,
+  DevNamespaceScope
```
Comment (joshuawilson) [inline src/app/create/environments/environments.module.ts]: KubernetesRestangularModule, NamespaceScope, DevNamespaceScope import not used

File: src/app/create/environments/environments.module.ts [hunk 12]
```diff
@@ -2,7 +2,7 @@ import { NgModule } from '@angular/core';
 import { CommonModule } from '@angular/common';
 import { HttpModule, Http } from '@angular/http';
 
-import { ToolbarModule, ToolbarConfig } from 'ngx-widgets';
+import { ToolbarModule, ToolbarConfig, TreeListModule } from 'ngx-widgets';
```
Comment (joshuawilson) [inline src/app/create/environments/environments.module.ts]: ToolbarConfig import not used

File: src/app/create/environments/environments.module.ts [hunk 13]
```diff
@@ -11,54 +11,39 @@ import {
...
+
+import {
+  KubernetesRestangularModule,
+  NamespaceScope,
+  DeploymentModule,
+  DevNamespaceScope
+} from 'fabric8-runtime-console';
+import { RestangularModule } from 'ng2-restangular';
+
+import { SwitchableNamespaceScope } from './../../shared/runtime-console/switchable-namepsace.scope';
```
Comment (joshuawilson) [inline src/app/create/environments/environments.module.ts]: import not used

File: src/app/create/pipelines/pipelines.component.ts [hunk 14]
```diff
@@ -1,12 +1,14 @@
...
-import { Observable } from 'rxjs';
 import { Component, OnInit } from '@angular/core';
 import { Router } from '@angular/router';
-import { AuthenticationService, UserService } from 'ngx-login-client';
 
+import { Observable } from 'rxjs';
+
+import { AuthenticationService, UserService } from 'ngx-login-client';
 import { ToolbarConfig, FilterConfig, FilterQuery, FilterEvent, Filter, SortEvent, SortField } from 'ngx-widgets';
+import { Space, SpaceService } from 'ngx-fabric8-wit';
```
Comment (joshuawilson) [inline src/app/create/pipelines/pipelines.component.ts]: import not used

File: src/app/shared/fabric8-ui-http.service.ts [hunk 15]
```diff
@@ -0,0 +1,36 @@
+import { Observable } from 'rxjs';
+import { WIT_API_URL } from 'ngx-fabric8-wit';
+import { HttpService, Broadcaster, SSO_API_URL } from 'ngx-login-client';
+import { Injectable, Inject, forwardRef } from '@angular/core';
```
Comment (joshuawilson) [inline src/app/shared/fabric8-ui-http.service.ts]: Broadcaster and forwardRef import not used

File: src/app/shared/runtime-console/fabric8-ui-onlogin.service.ts [hunk 16]
```diff
@@ -0,0 +1,22 @@
+import { LoginService } from './../login.service';
+import { OnLogin, OAuthConfigStore } from 'fabric8-runtime-console';
+import { Injectable } from '@angular/core';
+
+@Injectable()
+export class Fabric8UIOnLogin implements OnLogin {
```
Comment (joshuawilson) [inline src/app/shared/runtime-console/fabric8-ui-onlogin.service.ts]: This class is not correctly implementing OnLogin, it needs a "whenLoggedIn" property.

File: src/app/shared/runtime-console/fabric8-ui-onlogin.service.ts [hunk 17]
```diff
@@ -0,0 +1,22 @@
+import { LoginService } from './../login.service';
+import { OnLogin, OAuthConfigStore } from 'fabric8-runtime-console';
```
Comment (joshuawilson) [inline src/app/shared/runtime-console/fabric8-ui-onlogin.service.ts]: OAuthConfigStore import is not used.

Imports are not alphabetical

File: src/app/shared/runtime-console/space-namespace.scope.ts [hunk 18]
```diff
@@ -10,14 +10,11 @@ import { StaticNamespaceScope } from './static-namespace.scope';
 import { ObservableFabric8UIConfig } from './../../shared/config/fabric8-ui-config.service';
 import { Fabric8UIConfig } from './../../shared/config/fabric8-ui-config';
```
Comment (joshuawilson) [inline src/app/shared/runtime-console/space-namespace.scope.ts]: import not used

File: src/app/shared/runtime-console/fabric8-ui-onlogin.service.ts (hunk 16 above)
Comment (pmuir) [inline src/app/shared/runtime-console/fabric8-ui-onlogin.service.ts]: I don't think it does

**Pull request merged**